pkg/controller/service/servicecontroller.go

```diff
@@ -314,6 +319,14 @@ func (s *ServiceController) createLoadBalancerIfNeeded(key string, service *v1.S
 
 func (s *ServiceController) persistUpdate(service *v1.Service) error {
 	var err error
+
+	// Make a copy so we don't mutate the shared informer cache
+	copy, err := api.Scheme.DeepCopy(service)
```

Comment:
Was this actually mutating the cached item?  I expect a TODO for that to not do so.